packages/m2m-gateway/src/services/tenantService.ts

```diff
@@ -40,6 +40,20 @@ export function tenantServiceBuilder(clients: PagoPAInteropBeClients) {
         },
       };
     },
+    getTenant: async (
+      tenantId: TenantId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.Tenant> => {
+      logger.info(`Retrieving tenant with id ${tenantId}`);
+
+      const { data: tenant } =
+        await clients.tenantProcessClient.tenant.getTenant({
```

Comment:
Same as in #1806, we should add 404 tenant not found to the API spec because this method can return the tenantNotFound error that will passthrough